Refactor Migrations and Upgrades Documentation (#10169)
opensearch-project/documentation-website: _migration-assistant/migration-phases/reroute-source-to-proxy.md

---
layout: default
title: Reroute client traffic
nav_order: 3
parent: Migration phases
grand_parent: Migration Assistant for OpenSearch
permalink: /migration-assistant/migration-phases/reroute-source-to-proxy/
---

> # Reroute client traffic to Traffic Capture Proxy

Reviewer: ```suggestion
# Reroute client traffic to the Traffic Capture Proxy
```

> **Note:** This page is only relevant if you are using Capture-and-Replay to avoid downtime during a migration. If you are only performing backfill migration or can tolerate downtime, you can skip this step.

Reviewer: ```suggestion
**Note**: This page is only relevant if you are using Capture-and-Replay to avoid downtime during a migration. If you are only performing backfill migration or can tolerate downtime, you can skip this step.
```

{: .note}

> ## Capture proxy data replication

Reviewer: ```suggestion
## Capture Proxy data replication
```

> If you're interested in capturing live traffic during your migration, Migration Assistant includes an Application Load Balancer for routing traffic to the capture proxy and the target cluster. Upstream client traffic must be routed through the capture proxy in order to replay the requests later. Before using the capture proxy, remember the following:

Reviewer: ```suggestion
If you're interested in capturing live traffic during your migration, Migration Assistant includes an Application Load Balancer for routing traffic to the Capture Proxy and target cluster. Upstream client traffic must be routed through the Capture Proxy in order to replay the requests later. Before using the Capture Proxy, remember the following:
```

> * The layer upstream from the Application Load Balancer is compatible with the certificate on the Application Load Balancer listener, whether it's for clients or a Network Load Balancer. The `albAcmCertArn` in the `cdk.context.json` may need to be provided to ensure that clients trust the Application Load Balancer certificate.

Reviewer: ```suggestion
* The layer upstream from the Application Load Balancer is compatible with the certificate for the Application Load Balancer listener, whether it's for clients or a Network Load Balancer. The `albAcmCertArn` in the `cdk.context.json` may need to be provided to ensure that clients trust the Application Load Balancer certificate.
```

* If a Network Load Balancer is used directly upstream of the Application Load Balancer, it must use a TLS listener.
* Upstream resources and security groups must allow network access to the Migration Assistant Application Load Balancer.

To set up the capture proxy, go to the AWS Management Console and navigate to **EC2 > Load Balancers > Migration Assistant Application Load Balancer**. Copy the Application Load Balancer URL. With the URL copied, you can use one of the following options.

### If you are using **Network Load Balancer → Application Load Balancer → Cluster**

> 1. Ensure that ingress is provided directly to the Application Load Balancer for the capture proxy.

Reviewer: ```suggestion
1. Ensure that ingress is provided directly to the Application Load Balancer for the Capture Proxy.
```

2. Create a target group for the Migration Assistant Application Load Balancer on port `9200`, and set the health check to `HTTPS`.
3. Associate this target group with your existing Network Load Balancer on a new listener for testing.

> 4. Verify that the health check is successful, and perform smoke testing with some clients through the new listener port.

Reviewer: ```suggestion
4. Verify that the health check is successful and perform smoke testing with some clients through the new listener port.
```

5. Once you are ready to migrate all clients, detach the Migration Assistant Application Load Balancer target group from the testing Network Load Balancer listener and modify the existing Network Load Balancer listener to direct traffic to this target group.
6. Now client requests will be routed through the proxy (once they establish a new connection). Verify the application metrics.

### If you are using **Network Load Balancer → Cluster**

If you do not want to modify application logic, add an Application Load Balancer in front of your cluster and follow the **Network Load Balancer → Application Load Balancer → Cluster** steps. Otherwise:

1. Create a target group for the Application Load Balancer on port `9200` and set the health check to `HTTPS`.
2. Associate this target group with your existing Network Load Balancer on a new listener.

> 3. Verify that the health check is successful, and perform smoke testing with some clients through the new listener port.

Reviewer: ```suggestion
3. Verify that the health check is successful and perform smoke testing with some clients through the new listener port.
```

4. Once you are ready to migrate all clients, deploy a change so that clients hit the new listener.
   

### If you are **not using an Network Load Balancer**

If you're only using backfill as your migration technique, make a client/DNS change to route clients to the Migration Assistant Application Load Balancer on port `9200`.

### Kafka connection

> After you have routed the client based on your use case, test adding records against HTTP requests using the following steps:

Reviewer: ```suggestion
After you have routed the client based on your use case, test adding records against HTTP requests using the following steps.
```

In the migration console, run the following command:

```bash
console kafka describe-topic-records
```
{% include copy.html %}
   
Note the records in the logging topic.
   

> After a short period, execute the same command again and compare the increased number of records against the expected HTTP requests.

Reviewer: ```suggestion
After a short period, re-execute the same command and compare the increased number of records against the expected HTTP requests.
```

## Backfilling documents to the source cluster

> From the snapshot you created of your source cluster, you can begin backfilling documents into the target cluster. Once you have started this process, a fleet of workers will spin up to read the snapshot and reindex documents into the target cluster. This fleet of workers can be scaled to increased the speed at which documents are reindexed into the target cluster.

Reviewer: ```suggestion
From your source cluster snapshot, you can begin backfilling documents into the target cluster. Once you have started this process, a fleet of workers will spin up to read the snapshot and reindex documents into the target cluster. This fleet of workers can be scaled to increase the speed at which documents are reindexed into the target cluster.
```
